homeassistant/components/overkiz/entity.py

```diff
@@ -34,8 +34,32 @@ def __init__(
         self._attr_available = self.device.available
         self._attr_unique_id = self.device.device_url
 
+        if self.is_sub_device and not self.is_sub_device_default_naming:
+            # In case of sub entity without default naming, use the provided label as name
+            self._attr_name = self.device.label
```

Comment:
Is this the name of the device itself? That shouldn't be necessary because with entity naming, the device name is already automatically added in front of the entity name.

Reply:
And `self.device.label` is returned as name of the device in `device_info`. With entity naming if it should just be the device name, set it to `None`

Reply:
@balloob the issue here is that Overkiz devices can be a bit complex. Let me explain how it would look.

For example, the Atlantic heaters are one physical device (in the real world), but are represented as multiple devices via Overkiz. They share the same device url, with a different subsystemid (#2 etc). We decided to link them all to the same device in Home Assistant, since this reflects the real-world scenario the best. However, in the case of @nyroDev his device has 4 different temperature sensors, where the name of the sensor is included in the label.

It is hard to change the entity descriptions for this, since all state keys are exactly the same, there are just 3 different temperature entries. My suggestion would be for this situation to include the label before the name, just as before, when there is a multi device situation.

```json
    "label": "Temp RDC",
    "deviceURL": "io:\/\/XXXXXX\/2674179#11",
    "shortcut": false,
    "controllableName": "io:AtlanticPassAPCZoneTemperatureSensor",
```

@nyroDev I don't understand the `is_sub_device_default_naming` though, why is this necessary? The code for this addition is not very clear and very 'hacky' currently. Perhaps we can do this change in another PR?

Reply:
@iMicknl Looks good to me as it will revert back how it was before the PR and won't have much effect regarding new naming convention.
New push coming in a few minutes.